proto/src/determined/experiment/v1/searcher.proto

```diff
@@ -6,21 +6,49 @@ option go_package = "github.com/determined-ai/determined/proto/pkg/experimentv1"
 import "google/protobuf/struct.proto";
 
 // WorkloadKind defines the kind of workload that should be executed by trial runners.
-enum WorkloadKind {
-    // Denotes an unknown workload kind.
-    WORKLOAD_KIND_UNSPECIFIED = 0;
-    // Signals to a trial runner that it should run a training step.
-    WORKLOAD_KIND_RUN_STEP = 1;
+enum Unit {
+    // Denotes an unknown unit.
+    UNIT_UNSPECIFIED = 0;
+    // Unit type "Batches".
+    UNIT_BATCHES = 1;
+    // Unit type "Records".
+    UNIT_RECORDS = 2;
+    // Unit type "Epochs".
+    UNIT_EPOCHS = 3;
+}
+
+// TrainingLength represents a single runnable operation emitted by a searcher.
+message TrainingLength {
+    // This is the unit the TrainingLength is in terms of.
+    Unit unit = 1;
+    // This is the magnitude of the TrainingLength.
+    int32 units = 2;
```

Comment:
I'll change:
`TrainingLength` => `TrainingUnits`
`units` => `count`

Reasoning is that the docs have a concept now of `TrainingUnits` and this defines an external API so it's nice if they're the same, and just renaming units to count doesn't work that well since the count of a "TrainingLength` is still a little awkward.